drivers/kubevirt/kubevirt.go

```diff
@@ -507,3 +511,181 @@ func AddPVCsToVirtualMachine(vm kubevirtv1.VirtualMachine, pvcs []*corev1.Persis
+	vols, err := driver.GetVolumes(vmi)
+	if err != nil {
+		return false, err
+	}
+	if len(vols) == 0 {
+		return false, fmt.Errorf("no volumes found for VM [%s] in namespace [%s]", vmi.App.Key, vmi.App.NameSpace)
+	}
+
+	// Get the node where pod is scheduled
+	vms, err := GetAllVMsFromScheduledContexts([]*scheduler.Context{vmi})
+	if err != nil {
+		return false, err
+	}
+
+	vm := vms[0]
+	log.InfoD("pod name: %s, namespace: %s", vmi.App.Key, vmi.App.NameSpace)
+
+	vmNodeName, err := GetNodeOfVM(vm)
+	if err != nil {
+		return false, err
+	}
+	log.Infof("Pod [%s] is deployed on node [%s]", vmi.App.Key, vmNodeName)
+
+	// Keep a track of replicaset and consider this as the source of truth
+	volInspect, err := volumeDriver.InspectVolume(vols[0].ID)
+	if err != nil {
+		return false, fmt.Errorf("failed to inspect volume [%s]: %w", vols[0].ID, err)
+	}
+	globalReplicSet := volInspect.ReplicaSets
+	log.InfoD("Length of replicaset: %d", len(globalReplicSet))
+
+	// Check if the volume is mounted on the node
+	for _, vol := range vols {
+		// Check if the volume is mounted on the node
+		t := func() (interface{}, bool, error) {
+			volInspect, err := volumeDriver.InspectVolume(vol.ID)
+			if err != nil {
+				return "", false, fmt.Errorf("failed to inspect volume [%s]: %w", vol.ID, err)
+			}
+			nodeIpAttachedOn := volInspect.AttachedOn
+			nodeNameAttachedOn, err := node.GetNodeByIP(nodeIpAttachedOn)
+			if err != nil {
+				return "", false, fmt.Errorf("failed to get node name by IP [%s]: %w", nodeIpAttachedOn, err)
+			}
+			log.Infof("Volume [%s] is attached on node [%s]", vol.ID, nodeNameAttachedOn.Name)
+			if err != nil {
+				return "", false, fmt.Errorf("failed to check if volume [%s] is mounted on node [%s]: %w", vol.ID, vmNodeName, err)
+			}
+			if !(nodeNameAttachedOn.Name == vmNodeName) {
+				return "", true, fmt.Errorf("volume [%s] is not mounted on node [%s]", vol.ID, vmNodeName)
+			}
+			return "", false, nil
+		}
+		_, err = task.DoRetryWithTimeout(t, 1*time.Minute, 10*time.Second)
+		if err != nil {
+			return false, err
+		}
+		// Check if all the replicas are attached to the same node
+		volInspect, err := volumeDriver.InspectVolume(vol.ID)
+		if err != nil {
+			return false, fmt.Errorf("failed to inspect volume [%s]: %w", vol.ID, err)
+		}
+		replicaset := volInspect.ReplicaSets
+
+		// check if the replicaset size is same as the global replicaset size
+		if len(replicaset) != len(globalReplicSet) {
+			return false, fmt.Errorf("replicaset count mismatch for volume [%s]", vol.ID)
+		}
+		// The criteria to call the bind mount successful is to check if the replicaset of all volumes should be same
+		// check if the replicaset values is same as the global replicaset
+		// Here we are considering globalreplicaset to be source of truth for comparision
+		for _, rs := range replicaset {
+			found := false
+		outerLoop:
+			for _, rsNode := range rs.Nodes {
```

Comment:
need to verify that len(rs.Nodes) matches. This is the HA factor.

len(replicaSet) is aggr level which is usually 1 in all our tests.

Reply:
Yes done.